src/logger.ts

```diff
@@ -77,25 +112,36 @@ export class Logger implements ILogger {
     const prefix = this.config.appName ? `[${this.config.appName}] ` : '';
     message = `${prefix}${message}`;
 
-    try {
-      /* log to secondary logger (default is console) */
-      this.secondaryLogger[logLevel](message,
-        this.config.stringify ? stringify(details) : details
-      );
-    } catch (error) {
-      /* don't let custom logger errors stop our logger */
-      console.error('Error logging using custom logger passed into `genesys-cloud-client-logger`', { error, secondaryLogger: this.secondaryLogger, message, details, skipServer });
+    next(logLevel, message, details, messageOptions);
+  };
+
+  private logMessage = (
+    logLevel: LogLevel,
+    message: string,
+    details: any | undefined,
+    messageOptions: ILogMessageOptions
+  ): void => {
+    if (!messageOptions.skipSecondaryLogger) {
```

Comment:
I couldn't get this to trip, but isn't there a chance that `messageOptions` could be `undefined` here?